python/cudf/cudf/tests/test_scalar.py

```diff
@@ -289,3 +289,14 @@ def test_device_scalar_direct_construction(value):
         assert s.dtype == "object"
     else:
         assert s.dtype == dtype
+
+
+@pytest.mark.parametrize("value", SCALAR_VALUES)
+def test_construct_from_scalar(value):
+    value = cudf.utils.utils.to_cudf_compatible_scalar(value)
+    x = cudf.Scalar(1, value.dtype)
+    y = cudf.Scalar(x)
+    assert x.value == y.value
```

Comment:
Just noting this doesn't work for `NaN` but we're not testing it anyway right now.

Reply:
Oh wait, looks like we are testing for `NaN`.

Reply:
hang on.. there's something very wrong here:

```
    x = cudf.Scalar(1, value.dtype)
```

:P

Reply:
Fixed